### ISSUE TITLE
Messages not downloaded

### ISSUE THREAD

**Issue by ANEast** [thunderbird/thunderbird-android#3847]
I have come across an error that seems to be pervasive for the e-mails it affects.  I have 3 messages that it doesn't matter when I look at them, where I move them to (using K9 or Thunderbird / webmail on a desktop) they always appear as new notifications.  I have taken the log below and then resolved this on this occassion by clearing app data and then reloading the settings from backup.  The mails re-sync and I'm off again, until next time.  It appears that although K9 says it downloads the messages, it never does.  Any help would be gratefully appreciated, but as mentioned I have not got this problem for the time being after resetting K9.

### Expected behavior
- Notification of new e-mail messages.
- Notification cleared (by any of: swiping off / reading / marking read)
- Refresh the folder (manually or wait for next poll)
- No new notifications about the messages already "seen"

### Actual behavior
- For 3 specific messages, everytime K9 resyncs the folder, a new notification is generated.

### Steps to reproduce
Only possible with specific messages
1. Refresh folder
2. Read messages / clear notification
3. Refresh folder

### Environment
K-9 Mail version: 5.600

Android version: 6.0.1

Account type (IMAP, POP3, WebDAV/Exchange): IMAP (private server)

Please take some time to [retrieve logs](https://github.com/k9mail/k-9/wiki/LoggingErrors) and attach them here:

[k9log1812231756.txt](https://github.com/k9mail/k-9/files/2712614/k9log1812231756.txt)

**Comment by philipwhiuk** [comment]
Isn't this a duplicate of #3848 ?

**Comment by hettipeti** [comment]
The fault is for sure because of the following bug: #1422

**Comment by ANEast** [comment]
@philipwhiuk No, I don't think this is the same as #3848 - which is why I added them separately.  This bug is that even though it shows the messages, when the user reads them K9 doesn't store that state, and continuously notifies the user of new messages.  This error has cropped up again over the last few days.  5 messages dating back years are suddenly "unseen", and I can't get them to be "seen".  #3848 isn't to do with notifications and seen/unseen, it is which e-mails K9 actually pulls from the server.

@hettipeti I have no idea about the underlying source of the problem, but bug #1422 appears to be the same as my report for #3848  I apologise that I didn't find that (Although I did look).  However, if #1422 has been present for 2 1/2 years is it ever going to be fixed?
